JanSzala/SWAPI_client
# SWAPI_Client PR

#### Summary:
I've created basic scaffolding for the app. Created Main menu with animation, added downloading data from the server, simple parsing and checking for errors. Added dark mode (ability to change colors of the background and text). Added simple validation of internet connection and presentation of the downloaded data.

I've had trouble with presenting all the data, some of the data is sent as a link to the source on the server, and I couldn't quite figure out how to make it work. I've had trouble with adding swiftlint - it would ignore my swiftlint.yml file with rules, so I temporarily disabled it. Also right now only presentation of People details work. Im really embarrassed of how I build it, but I didn't find any other way. I know all my models are redundant, but I couldn't figure out way so they would inherit some of variables.

---
#### Tests:
- Open up the app.
   **Expected outcome:** You should see buttons animation - coming from the bottom of the screen and then adding 6 more buttons. 

- Open up the app, in the main menu tap on the button with the **cog** icon (in the center).
   **Expected outcome:** Mode should be changed depending on which mode was currently used. Backgrounds of the controllers and texts should be changed white/black. This should affect every controller in the app.

- Open up the app, tap in any of the buttons in the main menu.
   **Expected outcome:** You should be taken to the list containing names of the selected type, e.g. if you've chosen people, you should see list of people, if planets, list of planets and so on.

- Tap **People** button in the main menu and then tap any **person** on the list.
   **Expected outcome:** You should see this persons details (for now it only works for Peoples list)
